pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -114,7 +127,10 @@ func initDrainer(r *ReconcileNodeMaintenance, config *rest.Config) error {
 	r.drainer.Client = cs
 	r.drainer.DryRun = false
 
-	return nil
+	r.drainer.Out = writer{klog.Info}
+	r.drainer.ErrOut = writer{klog.Error}
+
+    return nil
```

Comment:
No need for `OnPodDeletedOrEvicted` ?

Reply:
OnPodDeletedOrEvicted callback is optional - the drainer only calls it if it has been set (field value is not nil). This differs from the Out fields - it is accessed without a check, so that the Out field is mandatory.  I could set the OnPodDeletedOrEvicted callback to log the event that a pod has been evicted. Do you request to do so?

Reply:
Yes please

Reply:
has been done in subsequent PR. please review.